### PR TITLE
Options page URL And Filtering 

### PR DESCRIPTION
**Pull request by thebritican**
Accomplishes:

1. working open now toggle
2. options page now filters
3. url has all filter stage in it
4. browser history between searches is preserved correctly
5. schedules table visualization is fixed and made into a component

linksf-filters-v2

Almost fixed schedules: has 0pm :P that should be 12
image

Using `Ramda.js` to do our piping. The convenience functions should also help some other code around linksf that has become unclear with the lack of default immutable updates in Javascript/ES6.

### When i began
selfie-0
[_GitHub Selfies_](https://github.com/thieman/github-selfies/)

### today
selfie-4
[_GitHub Selfies_](https://github.com/thieman/github-selfies/)

### CODE REVIEW

File: components/FilterBar/FilterBar.js [hunk 1]
```diff
@@ -4,21 +4,26 @@ import ToggleButton from '../ToggleButton'
 import s from './FilterBar.css'
 import Link from '../Link'
 
+const getNewQueryString = (queryString, showOpen) => (
+  showOpen ? `${queryString}&hours=open` : queryString.replace('&hours=open', '')
```
Comment (craig-day) [inline components/FilterBar/FilterBar.js]: will this break if the `queryString` coming in is empty?

File: lib/filterLocations.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    if (!service.schedules) return true
+    return (service.schedules).some(scheduleContainsCurrentTime)
+  }
+
+  return Object.values(services).some(serviceHasMatchingSchedule)
+}
+
+// Types!
+//
+// -- = example
```
Comment (craig-day) [inline lib/filterLocations.js]: nice Elm comment style there...

File: components/FilterBar/FilterBar.js (hunk 1 above)
Comment (thebritican) [inline components/FilterBar/FilterBar.js]: I don't believe so but i'll verify before merging

File: lib/filterLocations.js (hunk 2 above)
Comment (thebritican) [inline lib/filterLocations.js]: needed something to differentiate w/ `//` :)

File: components/FilterBar/FilterBar.js (hunk 1 above)
Comment (thebritican) [inline components/FilterBar/FilterBar.js]: verified that it's a'ight

**Pull request merged**